zcash_primitives/src/sapling/note_encryption.rs

```diff
@@ -669,6 +663,40 @@ mod tests {
         (ovk, ock, output)
     }
 
+    fn reencrypt_out_ciphertext(
+        ovk: &OutgoingViewingKey,
+        cv: &ValueCommitment,
+        cmu: &ExtractedNoteCommitment,
+        ephemeral_key: &EphemeralKeyBytes,
+        out_ciphertext: &[u8; OUT_CIPHERTEXT_SIZE],
+        modify_plaintext: impl Fn(&mut [u8; OUT_PLAINTEXT_SIZE]),
```

Comment:
It's so that we can reuse this function for a variety of different modifications. The caller provides a closure that implements the specific kind of modification that it is testing.